dataproc/submit_job_to_cluster.py

```diff
@@ -42,222 +61,230 @@ def get_region_from_zone(zone):
+                       filename):
+    """Submit the Pyspark job to the cluster (assumes `filename` was uploaded
+    to `bucket_name."""
     job_details = {
-        'projectId': project,
-        'job': {
-            'placement': {
-                'clusterName': cluster_name
-            },
-            'pysparkJob': {
-                'mainPythonFileUri': 'gs://{}/{}'.format(bucket_name, filename)
-            }
+        'placement': {
+            'cluster_name': cluster_name
+        },
+        'pyspark_job': {
+            'main_python_file_uri': 'gs://{}/{}'.format(bucket_name, filename)
         }
     }
-    result = dataproc.projects().regions().jobs().submit(
-        projectId=project,
-        region=region,
-        body=job_details).execute()
-    job_id = result['reference']['jobId']
-    print('Submitted job ID {}'.format(job_id))
+
+    result = dataproc.submit_job(
+        project_id=project, region=region, job=job_details)
+    job_id = result.reference.job_id
+    print('Submitted job ID {}.'.format(job_id))
     return job_id
 # [END dataproc_submit_pyspark_job]
 
 
 # [START dataproc_delete]
 def delete_cluster(dataproc, project, region, cluster):
-    print('Tearing down cluster')
-    result = dataproc.projects().regions().clusters().delete(
-        projectId=project,
-        region=region,
-        clusterName=cluster).execute()
+    """Delete the cluster."""
+    print('Tearing down cluster.')
+    result = dataproc.delete_cluster(
+        project_id=project, region=region, cluster_name=cluster)
     return result
 # [END dataproc_delete]
 
 
 # [START dataproc_wait]
 def wait_for_job(dataproc, project, region, job_id):
+    """Wait for job to complete or error out."""
     print('Waiting for job to finish...')
     while True:
-        result = dataproc.projects().regions().jobs().get(
-            projectId=project,
-            region=region,
-            jobId=job_id).execute()
+        job = dataproc.get_job(project, region, job_id)
         # Handle exceptions
-        if result['status']['state'] == 'ERROR':
-            raise Exception(result['status']['details'])
-        elif result['status']['state'] == 'DONE':
+        if job.status.State.Name(job.status.state) == 'ERROR':
+            raise Exception(job.status.details)
+        elif job.status.State.Name(job.status.state) == 'DONE':
             print('Job finished.')
-            return result
+            return job
 # [END dataproc_wait]
 
 
-# [START dataproc_get_client]
-def get_client():
-    """Builds an http client authenticated with the service account
-    credentials."""
-    dataproc = googleapiclient.discovery.build('dataproc', 'v1')
-    return dataproc
-# [END dataproc_get_client]
-
+def main(project_id,
+         zone,
+         cluster_name,
+         bucket_name,
+         pyspark_file=None,
+         create_new_cluster=True,
+         global_region=True):
+
+    # [START dataproc_get_client]
```

Comment:
Thoughts on refactoring this into a shared helper module? I imagine it could get confusing to have the same region tag in multiple files.

Reply:
Can you clarify? I'm not sure what you are suggesting. AFAIK, the global region is unique to Dataproc and the current get_region function doesn't (can't) parse out the "global" region from a standard zone uri.

Reply:
I mean this set of 8 lines or so is repeated several times throughout this PR. We could create a `dataproc_helpers.py` file that contains this as a method and use it from the sample files. (May need to add an empty `__init__.py` to use `from . import dataproc_helpers`)

Reply:
Now that I look at this more closely, the two `dataproc_get_client`s do different things.

I've filed a sample tracker request at devrel/samples/1703 (Googlers only, sorry public readers) and made the single job version a different region tag.

Reply:
Looks good. Thanks.